app/models/subnet.rb

```diff
@@ -213,6 +213,12 @@ def has_vlanid?
     self.vlanid.present?
   end
 
+  # Given a parameter name, return the (first) value, or nil if nothing found
+  def parameter_for_name(name)
```

Comment:
I believe `host_param` contains values from subnets as well, it does the whole classification. There's no `domain_param` or `os_param` either. The `parameter_for_name` is I think equivalent to `host_param` but it only searches for HostParameter objects.

Warnings during preview sounds as a great idea. Could you open RFE for that pls? I can imagine there would be more stuff we can warn about.